test/preamble-big.test.js

```diff
@@ -0,0 +1,66 @@
+import { beforeAll, describe, expect, it } from "@jest/globals";
+import Big from "big.js";
+globalThis.Big = Big;
+
+describe("Runtime tests for BigDecimal", function () {
+  beforeAll(function () {
+    return import("../preamble-big.js");
+  });
+
+  describe("Math.abs", function () {
+    it("handles decimal", function () {
+      expect(Math.abs(Big("-1.5"))).toEqual(Big("1.5"));
+      expect(Math.abs(Big("1.5"))).toEqual(Big("1.5"));
+      expect(Math.abs(Big("0"))).toEqual(Big("0"));
+    });
+
+    it("still handles non-decimal", function () {
+      expect(Math.abs(-1.5)).toEqual(1.5);
+      expect(Math.abs(1.5)).toEqual(1.5);
+      expect(Math.abs(0)).toEqual(0);
+    });
+  });
+
+  describe("Math.floor", function () {
+    it("handles decimal", function () {
+      expect(Math.floor(Big("-1.5"))).toEqual(Big("-2"));
+      expect(Math.floor(Big("1.5"))).toEqual(Big("1"));
+    });
+
+    it("still handles non-decimal", function () {
+      expect(Math.floor(-1.5)).toEqual(-2);
+      expect(Math.floor(1.5)).toEqual(1);
+    });
+  });
+
+  describe("Math.log10", function () {
+    it("handles decimal", function () {
```

Comment:
This test currently fails. I'll either change it or remove it before this is ready to merge.